src/components/DailyStatistics/utils.ts

```diff
@@ -0,0 +1,52 @@
+import { chain, sumBy } from "lodash";
+import { DailyStatistics, CampaignPolicy } from "../../types";
+
+type SummarisedTransactions = {
+  summarisedTransactionHistory: {
+    name: string;
+    category: string;
+    quantityText: string;
+  }[];
+  summarisedTotalCount: number;
+};
+
+export const countTotalTransactionsAndByCategory = (
+  response: DailyStatistics,
+  policies: CampaignPolicy[] | null
+): SummarisedTransactions => {
+  return chain(response.pastTransactions)
+    .groupBy("category")
+    .reduce<SummarisedTransactions>(
+      (prev, transactionsByCategory, key) => {
+        const totalQuantityInCategory = sumBy(
+          transactionsByCategory,
+          "quantity"
+        );
+        prev.summarisedTransactionHistory.push({
+          name: policies?.find(item => item.category === key)?.name ?? key,
+          category: key,
+          quantityText: formatQuantityText(
+            totalQuantityInCategory,
+            policies?.find(item => item.category === key)?.quantity.unit || {
+              type: "POSTFIX",
+              label: " qty"
+            }
+          )
+        });
+        prev.summarisedTotalCount += totalQuantityInCategory;
+        return prev;
+      },
+      { summarisedTransactionHistory: [], summarisedTotalCount: 0 }
+    )
+    .value();
+};
+
+export const formatQuantityText = (
```

Comment:
i've added the use of `toLocaleString()` to ensure thousands have a comma, i.e. 999,999,999 instead of 999999999. let me remove the extra one and add `toLocaleString()` in!